src/components/Asset/AssetBalance/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useEffect, useState } from "react"
+import {
+  useGetNearNativeBalance,
+  useGetNearNep141BalanceAccount,
+} from "src/hooks/useNearGetTokenBalance"
+import { useUser } from "src/providers/UserContext"
+import type { SwappableToken } from "src/types"
+import type { BaseTokenInfo } from "src/types/base"
+import { isBaseToken } from "src/utils"
+import { assert } from "src/utils/assert"
+import { balanceToDecimal } from "src/utils/balanceTo"
+
+type AssetBalanceProps = {
+  asset: SwappableToken
+}
+
+export const AssetBalance = ({ asset }: AssetBalanceProps) => {
+  const [balance, setBalance] = useState<bigint | null>(null)
+  const { userAddress } = useUser()
+
+  const { data: balanceData, mutate: mutateNearNep141BalanceAccount } =
+    useGetNearNep141BalanceAccount({ retry: true })
+  const { data: balanceNativeData, mutate: mutateNearNativeBalance } =
+    useGetNearNativeBalance({ retry: true })
```

Comment:
Here the same concern about usage of mutation over query. It looks like double work.